lib/cli/file-set-pipeline/configure.js

```diff
@@ -28,16 +28,16 @@ var Configuration = require('../configuration');
  * @param {CLI} cli
  */
 function configure(cli) {
-    var configuration = new Configuration({
-        'detectRC': cli.detectRC,
-        'file': cli.configPath,
-        'settings': cli.settings,
-        'plugins': cli.plugins,
-        'output': cli.output,
-        'cwd': cli.cwd
-    });
-
-    cli.configuration = configuration;
+    if (!cli.configuration) {
```

Comment:
Things will work fine (this is also related to your previous question), however, by wrapping in an if-statement no new 'congifuration' instance is created on subsequent 'run's.

You should be able to reproduce this by changing a lint rule value in .mdastrc (see mdast-lint for how to do that), E.g., setting maximum-line-length to 20. Theoretically, this change shouldn't start warning because the configuration is cached in memory and not reloaded! But, this isn't true if you change a pattern in the .mdastignore file as it isn't cached.

Did that help? :)